apps/worker/tasks/base.py

```diff
@@ -232,6 +234,67 @@ def _retry(self, countdown=None):
             if TestResultsFlow.has_begun():
                 TestResultsFlow.log(TestResultsFlow.UNCAUGHT_RETRY_EXCEPTION)
 
+    def _get_total_attempts(self) -> int:
+        """
+        Get total attempts including re-deliveries from visibility timeout expiration.
+
+        When visibility timeout expires, Redis re-queues the task but Celery treats
+        it as a new delivery (not a retry), so request.retries doesn't increment.
+        We track total attempts in task headers to catch these re-deliveries.
+
+        Returns:
+            Total number of attempts (retries + re-deliveries)
+        """
+        if not hasattr(self, "request"):
+            return 0
```

Comment:
I feel like this should return `None` instead no?

Reply:
No, because this is an int counter, and I don't think that None can be incremented. I think we want `0` in this case. Unless python hates that.